lib/puppet/parser/scope.rb

```diff
@@ -816,12 +845,12 @@ def effective_symtable(use_ephemeral)
   #
   # @param varname [String] The variable name to which the value is assigned. Must not contain `::`
   # @param value [String] The value to assign to the given variable name.
-  # @param options [Hash] Additional options, not part of api.
+  # @param options [Hash] Additional options, not part of api and no longer used.
   #
   # @api public
   #
-  def []=(varname, value, options = {})
-    setvar(varname, value, options = {})
```

Comment:
I just saw @hlindberg's comment. Are you still discussing this issue?

One question is whether this was accidentally broken recently, perhaps in a bad merge? As in, may be this did work correctly in 4.7, but was broken in between? But if it's one of those things that has been broken for a very long time, then I'm 👍  to your change.